docs(tutorial): Create GitHub action to sync object storage (#4586)
scaleway/docs-content: tutorials/cicd-github-action-object-storage-sync/index.mdx

content:
  h1: Using Github action to Sync your files with Object Storage
  paragraph: This page explains how to configure your github action to sync your files with an Object storage
tags: CI/CD Github Action Object Storage
categories:
  - object-storage
dates:
  validation: 2025-03-10
  posted: 2025-03-10
---

## Tutorial: Deploying your content to Scaleway Object Storage Using GitHub Actions

This tutorial will guide you through setting up a GitHub Action to deploy your Astro site to Scaleway Object Storage. We will use a GitHub Actions workflow to automate the deployment process whenever changes are pushed to the `main` branch.

## Prerequisites

- A GitHub repository with your Astro project.
- A Scaleway account with Object Storage configured.
- Scaleway API credentials (Access Key, Secret Key, Project ID, Organization ID).

## Step 1: Create the GitHub Action Workflow

1. In your GitHub repository, navigate to the `.github/workflows` directory. If it doesn't exist, create it.
2. Create a new file named `deploy.yml` in the `.github/workflows` directory.

## Step 2: Define the Workflow

> Copy and paste the following code into the `deploy.yml` file. This workflow will run on every push to the `main` branch, build your Astro site, and upload the built files to Scaleway Object Storage.

Reviewer: ```suggestion
Copy and paste the following code into the `deploy.yml` file. This workflow will run on every push to the `main` branch, build your Astro website, and upload the built files to Scaleway Object Storage.
```

```yaml
name: Upload to Scaleway Object Storage

on:
  push:
    branches:
      - main  # Change this to your default branch if different

jobs:
  upload:

>     environment: prd

Author: Indeed, this block code is not necessary

      - name: Install s3cmd
        run: sudo apt-get install -y s3cmd

      - name: Install CLI
        uses: scaleway/action-scw@v0
        with:
          version: v2.37.0

      - run: |
          scw object config get type=s3cmd > /home/runner/.s3cfg
          s3cmd --no-mime-magic --guess-mime-type sync ./dist/* s3://your_bucket_name/
        env:
          SCW_ACCESS_KEY: ${{ secrets.SCW_ACCESS_KEY }}
          SCW_SECRET_KEY: ${{ secrets.SCW_SECRET_KEY }}
          SCW_DEFAULT_PROJECT_ID: ${{ secrets.SCW_DEFAULT_PROJECT_ID }}
          SCW_DEFAULT_ORGANIZATION_ID: ${{ secrets.SCW_DEFAULT_ORGANIZATION_ID }}
```

## Step 3: Configure Secrets

In your GitHub repository, go to Settings > Secrets and variables > Actions.
Add the following secrets with your Scaleway credentials:

- SCW_ACCESS_KEY
- SCW_SECRET_KEY
- SCW_DEFAULT_PROJECT_ID
- SCW_DEFAULT_ORGANIZATION_ID

## Step 4: Commit and Push

Commit and push the `deploy.yml` file to your repository. This will trigger the GitHub Action to run on the main branch.

```bash
git add .github/workflows/deploy.yml
git commit -m "Add GitHub Action for deploying to Scaleway Object Storage"
git push origin main
```

> ### Notes:

Reviewer: ```suggestion

You can now access your website by clicking the **Website URL** from the **Settings** tab of your bucket in the [Scaleway console](https://console.scaleway.com/object-storage/buckets).
```

> #### Alternative Build Processes

Reviewer: ```suggestion
## Alternative build processes

```

> If your project uses a different build process, replace the `Install dependencies`and `Build the website` steps with the appropriate commands for your environment. 
> 
> Here are some examples:
> ```python

Reviewer: ```suggestion
If your project uses a different build process, replace the `Install dependencies`and `Build the website` steps with the appropriate commands for your environment. The code below shows an example of an alternative build process using [Astral](https://astral.sh/).

```yaml
```

      - name: Install uv
        uses: astral-sh/setup-uv@v5

      - name: Install the project
        run: uv sync --all-extras --dev
      - name: Build the project
        run: uv build

> ```

Reviewer: Add a this last part below

```

## Troubleshooting and additional resources

- If you encounter issues while using your API key with Scaleway Object Storage, refer to the [dedicated troubleshooting](/object-storage/troubleshooting/api-key-does-not-work/).

- Refer to the [official GitHub Actions documentation](https://docs.github.com/en/actions/writing-workflows/quickstart) for more information on how to create and run workflows.

- Learn how to better secure your deployment by properly [setting up IAM and Object Storage](/object-storage/api-cli/combining-iam-and-object-storage/)
```

          s3cmd --no-mime-magic --guess-mime-type sync ./dist/* s3://your_bucket_name/
        env:
          SCW_ACCESS_KEY: ${{ secrets.SCW_ACCESS_KEY }}
          SCW_SECRET_KEY: ${{ secrets.SCW_SECRET_KEY }}
          SCW_PROJECT_ID: ${{ secrets.SCW_PROJECT_ID }}
          SCW_ORGANIZATION_ID: ${{ secrets.SCW_ORGANIZATION_ID }}
```

Configure repository secrets

In your GitHub repository, go to Settings > Secrets and variables > Actions.
Add the following secrets with your Scaleway credentials:

- SCW_ACCESS_KEY
- SCW_SECRET_KEY
- SCW_PROJECT_ID
- SCW_ORGANIZATION_ID

Commit and push your changes

Run the commands below to commit and push the `deploy.yml` file to your repository. If you are working directly on your main branch, this will trigger the GitHub Action to run on the main branch.

```bash
git add .github/workflows/deploy.yml
git commit -m "Add GitHub Action for deploying to Scaleway Object Storage"
git push origin main
```

You can now access your website by clicking the **Website URL** from the **Settings** tab of your bucket in the [Scaleway console](https://console.scaleway.com/object-storage/buckets).

## Alternative build processes

> If your project uses a different build process, replace the `Install dependencies`and `Build the website` steps with the appropriate commands for your environment. The code below shows an example of an alternative build process using [Astral](https://astral.sh/).

Reviewer 2: ```suggestion
If your project uses a different build process, replace the `Install dependencies` and `Build the website` steps with the appropriate commands for your environment. The code below shows an example of an alternative build process using [Astral](https://astral.sh/).
```

```yaml
      - name: Install uv
        uses: astral-sh/setup-uv@v5

      - name: Install the project
        run: uv sync --all-extras --dev
      - name: Build the project
        run: uv build
```

## Troubleshooting and additional resources

- If you encounter issues while using your API key with Scaleway Object Storage, refer to the [dedicated troubleshooting](/object-storage/troubleshooting/api-key-does-not-work/).

- Refer to the [official GitHub Actions documentation](https://docs.github.com/en/actions/writing-workflows/quickstart) for more information on how to create and run workflows.

> - Learn how to better secure your deployment by properly [setting up IAM and Object Storage](/object-storage/api-cli/combining-iam-and-object-storage/)

Reviewer 2: ```suggestion
- Learn how to better secure your deployment by properly [setting up IAM and Object Storage](/object-storage/api-cli/combining-iam-and-object-storage/).
```
